src/backend/distributed/shared_library_init.c

```diff
@@ -980,6 +980,25 @@ RegisterCitusConfigVariables(void)
 		0,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.defer_drop_after_shard_split",
```

Comment:
I'm kind of wondering whether deferred drop should apply to create_distributed_table_concurrently (cc @aykut-bozkurt )

Reasons not to use it:
- The shard that gets split is on the coordinator andit  would be temporarily visible to users. While this is also true for regular splits in the query-from-any-node case, users who are administering the cluster will typically still connect to the coordinator.
- I would guess create_distributed_table_concurrently is less likely to suffer from co-located joins since it only works on a single table

Reason to use it:
- Select currently fails in the final stages, and would likely succeed with deferred drop

Reply:
Discussed it offline, we definitely need to do deferred drop, because even a simple transaction can deadlock:

begin;
-- takes an AccessShare lock on the shard on the worker (blocks DROP TABLE)
select * from test where x = 1;
-- takes an advisory lock on the coordinator (could get blocked )
update test set y = 3 where x = 1;
end;

Reply:
Discussed on the call. We do want deferred drop as it is very easy to get into deadlock even without any co-located joins.